x-pack/plugins/maps/public/components/map/feature_tooltip.js

```diff
@@ -5,12 +5,77 @@
  */
 
 import React, { Fragment } from 'react';
-import { EuiFlexGroup, EuiFlexItem, EuiButtonIcon } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiButtonIcon, EuiCallOut, EuiLoadingSpinner } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 
 
 export class FeatureTooltip extends React.Component {
 
+  state = {
+    properties: undefined,
+    loadPropertiesErrorMsg: undefined,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.prevLayerId = undefined;
+    this.prevFeatureId = undefined;
+    this._loadProperties();
+  }
+
+  componentDidUpdate() {
+    this._loadProperties();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  _loadProperties = () => {
+    this._fetchProperties({
+      nextFeatureId: this.props.tooltipState.featureId,
+      nextLayerId: this.props.tooltipState.layerId,
+    });
+  }
+
+  _fetchProperties = async ({ nextLayerId, nextFeatureId }) => {
+    if (this.prevLayerId === nextLayerId && this.prevFeatureId === nextFeatureId) {
+      // do not reload same feature properties
+      return;
+    }
+
+    this.prevLayerId = nextLayerId;
+    this.prevFeatureId = nextFeatureId;
+    this.setState({
+      properties: undefined,
+      loadPropertiesErrorMsg: undefined,
+    });
+
+    let properties;
+    try {
+      properties = await this.props.loadFeatureProperties({ layerId: nextLayerId, featureId: nextFeatureId });
+    } catch(error) {
+      if (this._isMounted) {
+        this.setState({
+          properties: [],
+          loadPropertiesErrorMsg: error.message
+        });
+      }
+      return;
+    }
+
+    if (this.prevLayerId !== nextLayerId && this.prevFeatureId !== nextFeatureId) {
```

Comment:
This is needed to handle out of order requests. For example, the user mouses over a point and the details are being fetch. Before the results return, the user mouses over another point, and `_fetchProperties` is called for the second point. When the first request returns, `_fetchProperties` needs to abort since the results are for the wrong feature.

Reply:
thanks!